packages/git/src/browser/git-view-contribution.ts

```diff
@@ -104,7 +104,16 @@ export class GitViewContribution extends AbstractViewContribution<GitWidget> {
         this.repositoryTracker.onGitEvent(event => {
             const { status } = event;
             const branch = status.branch ? status.branch : 'NO-HEAD';
-            const dirty = status.changes.length > 0 ? '*' : '';
+            let dirty = '';
+            if (status.changes.length > 0) {
+                if (this.hasConflicts(status.changes)) {
+                    dirty = '!';
+                } else if (this.allStaged(status.changes)) {
+                    dirty = '+';
+                } else {
+                    dirty = '*';
+                }
+            }
```

Comment:
It is possible to have all staged with a conflict, then we need to have `+!`. See here: https://github.com/theia-ide/theia/issues/1183#issuecomment-372049834